vendor/github.com/digitalbitbox/bitbox02-api-go/api/firmware/secp256k1.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firmware
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/digitalbitbox/bitbox02-api-go/util/errp"
+)
+
+func taggedSha256(tag []byte, msg []byte) []byte {
+	h := sha256.New()
+	tagHash := sha256.Sum256(tag)
+	h.Write(tagHash[:]) //nolint:errcheck
+	h.Write(tagHash[:]) //nolint:errcheck
+	h.Write(msg)        //nolint:errcheck
+	return h.Sum(nil)
+}
+
+func antikleptoHostCommit(hostNonce []byte) []byte {
+	return taggedSha256([]byte("s2c/ecdsa/data"), hostNonce)
+}
+
+// antikleptoVerify verifies that hostNonce was used to tweak the nonce during signature
+// generation according to k' = k + H(clientCommitment, hostNonce) by checking that
+// k'*G = signerCommitment + H(signerCommitment, hostNonce)*G.
+func antikleptoVerify(hostNonce, signerCommitment, signature []byte) error {
+	signerCommitmentPubkey, err := btcec.ParsePubKey(signerCommitment, btcec.S256())
+	fmt.Println("LOL===")
+	fmt.Println(hex.EncodeToString(hostNonce))
+	fmt.Println(hex.EncodeToString(signerCommitment))
+	fmt.Println(hex.EncodeToString(signature))
```

Comment:
OOOPS :joy:

Reply:
https://github.com/digitalbitbox/bitbox02-api-go/pull/49

Reply:
wait, I'm pretty sure I didn't see any lols in https://github.com/digitalbitbox/bitbox02-api-go/pull/48... how did it get there? :see_no_evil:

Reply:
i added a unit test as you requested, and then merged it without re-requesting a review. serves me right! :sweat_smile:

Reply:
maybe we should add a "lol" linter. THAT would be one linter I'd never complain about :rofl:

Reply:
That would be so helpful! Was not the first and probably not the last time ;)